game/scoring.cpp

```diff
@@ -0,0 +1,129 @@
+#include "yahtzee.h"
+#include <algorithm>
+// Util functions
+int sum(int die[5]) {
+    int sum = 0;
+    for (int i = 0; i < 5; ++i)
+    {
+        sum += die[i];
+    }
+    return sum;
+}
+
+//Funtions for scoring yahtzee hand
+
+// Out is an array initialized to zero in which to hold the count
+int sortDie(int die[5], int out[6]) {
+    for (int i = 0; i < 5; ++i)
+    {
+        out[die[i]-1] += 1;
+    }
+    return 0;
+}
+
+/*
+    scores x points for each die showing x pips
+*/
+int scoreNumber(int die[5], int x) {
+    int score = 0;
+    for (int i = 0; i < 5; ++i)
+    {
+        if (die[i] == x)
+        {
+            score += x;
+        }
+    }
+    return score;
+}
+
+/*
+    scores the total pips shown on all die if at least
+    x are the same rank. 
+*/
+int scoreXOfAKind(int die[5], int x) {
+    int ranks[6] = {0};
+    sortDie(die, ranks);
+    for (int i = 0; i < 6; ++i)
+    {
+        if(ranks[i] >= x) {
+            return sum(die);
+        }
+    }
+    return 0;
+}
+
+/*
+    scores 25 if the roll has 3 dice showing one rank and 
+    2 showing another. 
+*/
+int scoreFullHouse(int die[5]) {
+    int ranks[6] = {0};
+    sortDie(die, ranks);
+    int three_house = 0;
+    int two_house = 0;
+    for (int i = 0; i < 6; ++i)
+    {
+        if(ranks[i] == 2) {
+            two_house = 1;
+        }
+        else if (ranks[i] == 3)
+        {
+            three_house = 1;
+        }
+
+    }
+    if ((two_house == 1) & (three_house == 1))
+    {
+        return 25;
+    }
+    return 0;
+}
+
+/*
+    If cutoff die show consecutive ranks scores 30 if the cutoff is 
+    4 scores 40 if the cutoff is 5. 
+*/
+int scoreStraight(int die[5], int cutoff) {
```

Comment:
const ref to a vector for die. Also, add a parameter that's the score earned, don't compute it from the cutoff.